### PR TITLE
Defining groups ability

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/models/components/course/groups_ability_component.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def define_permissions
+    allow_staff_manage_groups if user
+
+    super
+  end
+
+  private
+
+  def allow_staff_manage_groups
+    can :manage, Course::Group, course_staff_hash
```
Comment (lowjoel) [inline app/models/components/course/groups_ability_component.rb]: how about the owner of the group?

Comment (allenwq) [inline app/models/components/course/groups_ability_component.rb]: I almost forgot that groups have roles ! ><

**Comment by allenwq** [comment]
@lowjoel Got some updates here.

**Comment by lowjoel** [comment]
This can work first. If there's other things necessary, we can do later.

**Pull request merged**